common/network_manager_test.go

```diff
@@ -199,3 +202,77 @@ func TestOnRequestPausedBlockedIPs(t *testing.T) {
 		})
 	}
 }
+
+func TestNetworkManagerEmitRequestResponseMetricsTimingSkew(t *testing.T) {
+	t.Parallel()
+
+	now := time.Now()
+	type tm struct{ ts, wt time.Time }
+	tests := []struct {
+		req, res, wantReq, wantRes tm
+	}{
+		{
+			req:     tm{ts: now, wt: now},
+			res:     tm{ts: now, wt: now},
+			wantReq: tm{wt: now},
+			wantRes: tm{wt: now},
+		},
+		{
+			req:     tm{ts: now, wt: now},
+			res:     tm{ts: now.Add(time.Minute)},
+			wantReq: tm{wt: now},
+			wantRes: tm{wt: now.Add(time.Minute)},
+		},
+		{
+			req:     tm{ts: now.Add(-time.Hour), wt: now},
+			res:     tm{ts: now.Add(-time.Hour).Add(time.Minute)},
+			wantReq: tm{wt: now},
+			wantRes: tm{wt: now.Add(time.Minute)},
+		},
+		{
+			req:     tm{ts: now.Add(time.Hour), wt: now},
+			res:     tm{ts: now.Add(time.Hour).Add(time.Minute)},
+			wantReq: tm{wt: now},
+			wantRes: tm{wt: now.Add(time.Minute)},
+		},
+		{
+			req:     tm{ts: now, wt: now.Add(-time.Hour)},
+			res:     tm{ts: now.Add(time.Minute)},
+			wantReq: tm{wt: now.Add(-time.Hour)},
+			wantRes: tm{wt: now.Add(-time.Hour).Add(time.Minute)},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run("", func(t *testing.T) {
```

Comment:
I left it as it didn't make sense to give it a name. I'm open to suggestions if you have any ideas about the names.

Reply:
IMO The subtest's name should describe what's going on, otherwise it's difficult to quickly understand what's happening. It will also help if something goes wrong in the tests, as atm this is what we get:

```
--- FAIL: TestNetworkManagerEmitRequestResponseMetricsTimingSkew (0.00s)
    --- FAIL: TestNetworkManagerEmitRequestResponseMetricsTimingSkew/#04 (0.00s)
```

I think they can just be something like:

`expected req & resp wt not affected when they're the same`

`expected req & resp wt affected when req wt is skewed`

etc

Reply:
Thanks for the suggestion. Although I agree with the benefits of naming in general, I think naming them wouldn't be much beneficial in this case. Other than adding unnecessary verbosity while you can read what's going on in the test case fields :(

Reply:
Then again: b84af436 :)